contributors/design-proposals/vertical-pod-autoscaler.md

```diff
@@ -0,0 +1,486 @@
+Vertical Pod Autoscaler
+=======================
+Vertical Pod Autoscaler (VPA, aka. "rightsizing" or "autopilot") is an
+infrastructure service that automatically sets resource requirements of Pods and
+dynamically adjusts them in runtime, based on analysis of historical resource
+utilization, amount of resources available in the cluster and real-time events,
+such as OOMs.
+
+- [Introduction](#introduction)
+  - [Background](#background)
+  - [Purpose](#purpose)
+  - [Related features](#related-features)
+- [Requirements](#requirements)
+  - [Functional](#functional)
+  - [Availability](#availability)
+  - [Extensibility](#extensibility)
+- [Design](#design)
+  - [Overview](#overview)
+  - [Architecture overview](#architecture-overview)
+  - [API](#api)
+  - [Admission Controller](#admission-controller)
+  - [Life of the Pod](#life-of-the-pod)
+  - [Recommender](#recommender)
+  - [Updater](#updater)
+  - [Recommendation model](#recommendation-model)
+  - [History Storage](#history-storage)
+  - [Open questions](#open-questions)
+- [Alternatives considered](#alternatives-considered)
+  - [Pods point at VPA](#pods-point-at-vpa)
+  - [VPA points at Deployment](#vpa-points-at-deployment)
+  - [Actuation using the Deployment update mechanism](#actuation-using-the-deployment-update-mechanism)
+  
+------------
+Introduction
+------------
+
+### Background ###
+* [Compute resources](http://kubernetes.io/docs/user-guide/compute-resources/)
+* [Resource QoS](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-qos.md)
+* [Admission Controllers](http://kubernetes.io/docs/admin/admission-controllers/)
+
+### Purpose ###
+Vertical scaling has two objectives: improving utilization of compute resources
+and reducing the maintenance cost, by automating configuration of resource
```

Comment:
Improving ease of use is the most important objective. Users coming from IaaS aren't used to setting container-level resources. Neither are users coming from a PaaS or FaaS with vertical scaling. And, TBH, we don't have good tools or guidance for people to figure out how to set resource values today. Additionally, I don't want Helm charts or Templates to need to come in tshirt sizes (S/M/L) or to plumb through cpu and memory parameters.